app/src/lib/multi-commit-operation.ts

```diff
@@ -0,0 +1,57 @@
+import { Branch } from '../models/branch'
+import {
+  ChooseBranchStep,
+  conflictSteps,
+  MultiCommitOperationStepKind,
+} from '../models/multi-commit-operation'
+import { Popup, PopupType } from '../models/popup'
+import { TipState } from '../models/tip'
+import { IMultiCommitOperationState, IRepositoryState } from './app-state'
+
+/**
+ * Setup the multi commit operation state when the user needs to select a branch as the
+ * base for the operation.
+ */
+export function getMultiCommitOperationChooseBranchStep(
+  state: IRepositoryState,
+  initialBranch?: Branch | null
+) {
```

Comment:
Maybe make the return value explicit?
```suggestion
): ChooseBranchStep {
```